tests/future/streams.rs

```diff
@@ -0,0 +1,50 @@
+use futures::{future::join_all, stream::FuturesUnordered, StreamExt};
+use shuttle::{
+    check_random,
+    future::{block_on, spawn},
+};
+
+fn futures_unordered_collect() {
+    block_on(async {
+        let tasks = (0..100).map(|_| spawn(async move {})).collect::<FuturesUnordered<_>>();
+
+        let _ = tasks.collect::<Vec<_>>().await;
+    });
+}
+
+#[test]
+fn collect_empty_tasks() {
+    check_random(futures_unordered_collect, 100);
+}
+
+fn futures_unordered_next() {
+    block_on(async {
+        let mut tasks = (0..100).map(|_| spawn(async move {})).collect::<FuturesUnordered<_>>();
+
+        while let Some(result) = tasks.next().await {
```

Comment:
Not sure this test really adds anything new, since the above collect essentially does the same thing, right?

Reply:
Hmm, kinda, the internals of `Next` and `Collect` are a bit different. I think it is fine to have both.